include/tack/float4.h

```diff
@@ -5,7 +5,15 @@
 #include <cstdint>
 #include <x86intrin.h>
 
+#include "common.h"
+
 namespace tack {
+template <typename in_t, typename out_t>
+inline out_t bitcast(const in_t & x)
+{
+    return *reinterpret_cast<const out_t*>(&x);
```

Comment:
Interesting note, I believe this also works because references use just pointers under the hood but have nice value semantics.

```cpp
return reinterpret_cast<const out_t&>(x);
```